### PR TITLE
Issue#1259 store display image

### LINKED ISSUE #1259: Make sure we use the same system to store and display image
(URL vs Reference firestorage)
- [x] Make some research
- [x] Make a decision on what to use
- [ ] Implement it everywhere

- [ ] Manage default images the same way everywhere
- [ ] use  this way : `logo/xxxx-xxxx-xxxx-xxxx.jpg` (Reference firestorage) to store images ( and the front recreate url from this path) 
- [ ] Add documentation on notion.so regarding file upload, cropper, various image sizes and ratios
- [x] Do not store default image path (assets/xxxxx) on firebase
- [x] Use `sanitizeFileName` (file-sanitizer.ts) to prevent filename collisions
- [ ] If possible, use `resourcesRatios` and `resourcesSizes` ( draw.io ) to store image meta datas (especially for movie `promotional elements`)

### PR DESCRIPTION
**Pull request by jenna1k**
resolve #1259

### CODE REVIEW

File: libs/movie/src/lib/movie/form/promotional-elements/promotional-elements.component.html [hunk 1]
```diff
@@ -2,22 +2,10 @@
...
-        </ng-template>
-        <mat-card-actions>
-          <button mat-raised-button color="warn" (click)="promotionalElements.removeImage(i)">
-            <mat-icon>delete_circle</mat-icon> Remove
-          </button>
-        </mat-card-actions>
-      </mat-card>
-    </div>
+    <drop-cropper [ratio]="4/3" storagePath="promotionalElements" [formControlName]="i" [useChangePic]="false" [useDelete]="false"></drop-cropper>
+    <button mat-raised-button color="warn" (click)="promotionalElements.removeImage(i)">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/promotional-elements/promotional-elements.component.html]: I think Mathilde said "no more mat-raised-button" use "mat-flat-button" instead :)

File: libs/ui/src/lib/avatar-list/avatar-list.component.html [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <img *ngIf="!!url; else unknown" [src]="url">
   </ng-container>
   <ng-container *ngFor="let ref of _refs">
-    <img *ngIf="!!ref; else unknown" [storageRef]="ref">
+    <img *ngIf="!!ref; else unknown" [imgRef]="ref">
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/avatar-list/avatar-list.component.html]: As you manage that internally, you can get rid of the `*ngIf="!!ref"` and the `ng-template` below to do : 
```
<img [imgRef]="ref" [placeholderUrl]="placeholderLogo" alt= "....">
```

File: libs/ui/src/lib/cropper/cropper/cropper.component.ts [hunk 3]
```diff
@@ -33,18 +40,23 @@ function b64toBlob(data: string) {
 
 
 /** Check if the path is a file path */
-function isFile(path: string): boolean {
-  if (!path) {
+function isFile(path: any): boolean {
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/cropper/cropper/cropper.component.ts]: Type your parameters: 
```typescript
function isFile(ref: ImgRef = {}): boolean {
  if (!path.ref) {
    return false;
  }
  ...
}
```

File: libs/ui/src/lib/cropper/cropper/cropper.component.ts [hunk 4]
```diff
@@ -81,8 +93,10 @@ export class CropperComponent implements ControlValueAccessor{
   }
 
   @Input() storagePath: string;
+  @Input() useChangePic?: boolean;
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/cropper/cropper/cropper.component.ts]: This is not a good name. I don't understand what it does.
Also add a comment on top of your inputs : 
```typescript
/** Do ... */
@Input() useChangePic?: boolean;
```

File: libs/ui/src/lib/cropper/cropper/cropper.component.ts [hunk 5]
```diff
@@ -92,29 +106,34 @@ export class CropperComponent implements ControlValueAccessor{
...
+      this.folder = this.storagePath;
+      this.nextStep('drop');
+    }
   }
 
   // update the parent form field when there is change in the component (component -> parent)
   registerOnChange(fn: any): void {
-    this.uploaded = (ref: string) => fn(ref);
-    this.deleted = () => fn(null);
+    console.log('register on change');
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/cropper/cropper/cropper.component.ts]: Remove console.log

File: libs/ui/src/lib/cropper/cropper/cropper.component.ts [hunk 6]
```diff
@@ -92,29 +106,34 @@ export class CropperComponent implements ControlValueAccessor{
...
+    this.uploaded = (ref: ImgRef) => fn(ref);
+    this.deleted = () => fn({
+      url: '',
+    ref: '',
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/cropper/cropper/cropper.component.ts]: Format

File: libs/ui/src/lib/cropper/cropper/cropper.component.ts [hunk 7]
```diff
@@ -33,18 +40,23 @@ function b64toBlob(data: string) {
...
+  const part = path.ref.split('.');
   const last = part.pop();
-  return part.length === 1 && !last.includes('/');
+  return part.length >= 1 && !last.includes('/');
 }
 
-
-
-
+export function createImgRef(ref: Partial<ImgRef> = {}): ImgRef {
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/cropper/cropper/cropper.component.ts]: I'm not sure that you really need this function.

File: libs/ui/src/lib/cropper/cropper/cropper.component.ts [hunk 8]
```diff
@@ -160,9 +180,18 @@ export class CropperComponent implements ControlValueAccessor{
 
   goToShow() {
     this.url$ = this.ref.getDownloadURL();
-    this.ref.getMetadata().toPromise()
-    .then(meta => this.uploaded(meta.fullPath));
+    combineLatest([this.url$, this.ref.getMetadata()]).pipe(
+      finalize(() => console.log('completed'))
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/cropper/cropper/cropper.component.ts]: Remove finalize

File: libs/ui/src/lib/cropper/cropper/cropper.component.ts [hunk 9]
```diff
@@ -160,9 +180,18 @@ export class CropperComponent implements ControlValueAccessor{
 
   goToShow() {
     this.url$ = this.ref.getDownloadURL();
-    this.ref.getMetadata().toPromise()
-    .then(meta => this.uploaded(meta.fullPath));
+    combineLatest([this.url$, this.ref.getMetadata()]).pipe(
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/cropper/cropper/cropper.component.ts]: Add a comment on top: `// Observable completed once both requests are completed`. Like that we know it's not a "unhandled subscription"

File: libs/ui/src/lib/cropper/cropper/cropper.component.ts [hunk 10]
```diff
@@ -160,9 +180,18 @@ export class CropperComponent implements ControlValueAccessor{
...
+      finalize(() => console.log('completed'))
+    ).subscribe(([url, meta]) => {
+      const newRef: ImgRef = createImgRef({
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/cropper/cropper/cropper.component.ts]: Here createImgRef does nothing.

File: libs/ui/src/lib/avatar-list/avatar-list.component.html (hunk 2 above)
Comment (jenna1k) [reply libs/ui/src/lib/avatar-list/avatar-list.component.html]: should I change ng-container to ng-template?

File: libs/ui/src/lib/cropper/cropper/cropper.component.ts (hunk 4 above)
Comment (jenna1k) [reply libs/ui/src/lib/cropper/cropper/cropper.component.ts]: it's for disable fileuploader button in the last step

File: libs/ui/src/lib/cropper/cropper/cropper.component.ts [hunk 11]
```diff
@@ -180,14 +173,15 @@ export class CropperComponent implements ControlValueAccessor{
...
       finalize(() => console.log('completed'))
     ).subscribe(([url, meta]) => {
-      const newRef: ImgRef = createImgRef({
+      const newRef: ImgRef = {
         url: url,
         ref: meta.fullPath,
         originalRef: ''
-      });
+      };
       this.uploaded(newRef);
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/cropper/cropper/cropper.component.ts]: You can do : 
```typescript
this.uploaded({ url, ref: meta.fullPath, originalRef: '' })
```

File: libs/ui/src/lib/avatar-list/avatar-list.component.html (hunk 2 above)
Comment (GrandSchtroumpf) [reply libs/ui/src/lib/avatar-list/avatar-list.component.html]: Remove the `ng-template`

**Pull request merged**